### DISCUSSION TITLE
Hop calculation issues

### DISCUSSION THREAD

**Discussion by jimdbr** [Brewtarget/brewtarget#596] [General]
I am using the latest development version of Brewtarget. I am having issues with hops additions. The calculated IBU is not updated correctly until after I leave the hop tab and return. The numbers are sometimes bogus. A couple of examples:

I added 28g of Cascade pellets for 60 minutes and had 0 IBU.

I added 28g of Eldorado pellets for 10 minutes and had a calculated IBU of approximately 43,000. I love hops but this might be excessive.

I added Galaxy pellets with an alpha of 12 to the database. When I include it in a recipe it comes in as leaf hops instead of pellets.

I don't have the previous version of Brewtarget available so I need to head over to Brewers Friend for today's brew. I will play with Brewtarget later to see if any of this is reproducible.

**Comment by mikfire** [comment]
I know what this is. Signals are being sent before the cache is updated. This is a side effect of the versioning changes.

I'm not so clear on how best to fix it. The right information is being written to the DB, so there is no permanent harm being done.

I've also lost internet access until tomorrow, so any brilliant solution will wait until tomorrow.

Mik

**Comment by mikfire** [comment]
Just submitted the pull request that fixes 90% of this problem. The equipment editor is an outstanding issue. It works, but it is ugly.

**Comment by jimdbr** [comment]
I built the latest development version and tried my recipe again. Hop updates work OK now. To fix the Galaxy leaf vs pellet issue I had to delete Galaxy from my database and recipe and add it again. I will do more testing later.

Thanks.

**Reply by mikfire** [reply]
Oh. Sorry. Forgot to put that into the commit log. 

That problem was a straight up bug. Hop::setForm() hadn't been updated to the new paradigm. That got fixed as I was fixing everything else in this PR.

Test it, of course, but I believe it to be fixed.

**Comment by jimdbr** [comment]
It is fixed.
